### PR TITLE
Create add command for doctor

### LINKED ISSUE #56: [DEV] As a doctor, I want to add a new colleague (doctor) details
* maybe try to separate the storage from patients

### PR DESCRIPTION
**Pull request by yyhangz**
### Summary

- Resolves #56

### Minor changes

- Refactor the model stub class in the test folder to `CommandTestUtil` for more reusablility

### Pictures / Gif

image

image

### Checks

- [X] Link PR to issue
- [X] Add Reviewer
- [x] Pass CI

### CODE REVIEW

File: src/main/java/gomedic/logic/parser/CliSyntax.java [hunk 1]
```diff
@@ -5,13 +5,16 @@
...
-    /* Prefix definitions for a person */
+    /* General prefixes */
     public static final Prefix PREFIX_NAME = new Prefix("n/");
     public static final Prefix PREFIX_PHONE = new Prefix("p/");
     public static final Prefix PREFIX_EMAIL = new Prefix("e/");
     public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
     public static final Prefix PREFIX_TAG = new Prefix("ta/");
 
+    /* Prefix definitions for a doctor */
+    public static final Prefix PREFIX_DEPARTMENT = new Prefix("s/");
```
Comment (simonjulianl) [inline src/main/java/gomedic/logic/parser/CliSyntax.java:16]: Is it better to use `de/` instead, `/s` is not really intuitive

File: src/main/java/gomedic/model/person/UniqueAbstractPersonList.java [hunk 2]
```diff
@@ -29,6 +29,21 @@
     private final ObservableList<T> internalUnmodifiableList =
             FXCollections.unmodifiableObservableList(internalList);
 
+    /**
+     * Returns a new available id.
+     * If it's an empty list, return id 1.
+     */
+    public int getNewId() {
```
Comment (simonjulianl) [inline src/main/java/gomedic/model/person/UniqueAbstractPersonList.java:36]: Can change this to these pieces of code 
```suggestion
    public int getNewId () {
        if (internalList.size() == 0) {
            return 1;
        }

        int counter = 1;
        // Find first id that does not exist in the list starting from D001
        for (Iterator<Integer> it = internalList
                .stream()
                .map(entry -> entry.getActivityId().getIdNumber())
                .sorted()
                .iterator(); it.hasNext(); counter++) {
            int id = it.next();
            if (id != counter) {
                return counter;
            }
        }

        // all the id are in sequences, make a new one
        if (counter <= MAX_CAPACITY) {
            return counter;
        }

        throw new MaxAddressBookCapacityReached();
    }
```

File: src/main/java/gomedic/model/person/UniqueAbstractPersonList.java [hunk 3]
```diff
@@ -29,6 +29,21 @@
...
+        try {
+            return internalList
+                    .get(internalList.size() - 1)
+                    .getId()
+                    .getIdNumber() + 1;
+        } catch (Exception e) {
+            return 1;
+        }
+    }
```
Comment (simonjulianl) [inline src/main/java/gomedic/model/person/UniqueAbstractPersonList.java:45]: Maybe we should change this according to your suggestion
```suggestion
    public int getNewId() {
        try {
            return internalList
                    .get(internalList.size() - 1)
                    .getId()
                    .getIdNumber() + 1;
        } catch (Exception e) {
            return 1;
        }
    }
    public int getNewId() {
	        if (internalList.size() == 0) {
	            return 1;
	        }
	
	        int counter = 1;
	        // Find first id that does not exist in the list starting from D001
	        for (Iterator<Integer> it = internalList
	                .stream()
	                .map(entry -> entry.getActivityId().getIdNumber())
	                .sorted()
	                .iterator(); it.hasNext(); counter++) {
	            int id = it.next();
	            if (id != counter) {
	                return counter;
	            }
	        }
	
	        // all the id are in sequences, make a new one
	        if (counter <= MAX_CAPACITY) {
	            return counter;
	        }
	
	        throw new MaxAddressBookCapacityReached();
	    }
```

File: src/main/java/gomedic/logic/parser/CliSyntax.java (hunk 1 above)
Comment (yyhangz) [reply src/main/java/gomedic/logic/parser/CliSyntax.java:16]: yeap i think de/ would make more sense, shall update it!

File: src/main/java/gomedic/model/person/UniqueAbstractPersonList.java (hunk 3 above)
Comment (yyhangz) [reply src/main/java/gomedic/model/person/UniqueAbstractPersonList.java]: Sorry i didnt realize that u were working on the id part as well, I added my own version of it too for the doctor class, lemme know if i shld change it

**Pull request merged**